### PR TITLE
convert url/email features to object before categorical

### LINKED ISSUE #3366: Imputer cannot impute generated features from EmailFeaturizer if missing values are present
#### Code Sample, a copy-pastable example to reproduce your bug.

```python
import pandas as pd
import woodwork as ww
from evalml.pipelines import RegressionPipeline
from evalml.pipelines.components import EmailFeaturizer, Imputer
import pytest

X = pd.DataFrame({"email": ["me@email.com", "foo@bar.org", "baz@foo.gov", None],
                  "number": [1, 2, 3, 4]})
y = pd.Series([1, 2, 1, 3])

pl = RegressionPipeline([EmailFeaturizer, Imputer])

with pytest.raises(AttributeError, match="'bool' object has no attribute 'transpose'"):
    pl.fit(X, y)

```

This is related to the pandas 1.4.0 upgrade. Pandas 1.4.0 will create use the nullable `<NA>` type for missing values in categorical features! Before it was just numpy `NaN`.

### Pandas 1.4.0
image

### Pandas 1.3.5
image

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #3366 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/pipelines/components/transformers/preprocessing/transform_primitive_components.py [hunk 1]
```diff
@@ -73,12 +73,19 @@ def transform(self, X, y=None):
...
         es = self._make_entity_set(X_ww)
         features = ft.calculate_feature_matrix(features=self._features, entityset=es)
-        features.set_index(X_ww.index, inplace=True)
 
-        X_ww = X_ww.ww.drop(self._columns)
+        # Convert to object dtype so that pd.NA is converted to np.nan
+        # until sklearn imputer can handle pd.NA in release 1.1
+        # FT returns these as string types, currently there isn't much difference
+        # in terms of performance between object and string
+        # see https://pandas.pydata.org/docs/user_guide/text.html#text-data-types
```
Comment (freddyaboulton) [inline evalml/pipelines/components/transformers/preprocessing/transform_primitive_components.py:81]: Need to perf test this

**Review by chukarsten — APPROVED** [review]
LGTM! Only thing i might consider is possibly just adding the columns to a module level dataframe for testing instead of making a new one, but thats not a big deal.

**Review by jeremyliweishih — APPROVED** [review]
very nice!

**Review by bchen1116 — APPROVED** [review]
Perf tests look great!

**Pull request merged**